### PR TITLE
Astro 2.10 article

### PR DESCRIPTION
**Pull request by matthewp**
Direct link: https://deploy-preview-757--astro-www-2.netlify.app/blog/astro-2100/

### CODE REVIEW

File: src/content/blog/astro-2100.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+While you're at it, upgrade any `@astrojs/*` integrations and adapters you have installed, too!
+
+# Persistent UI in View Transitions
+
+Our first release of the `<ViewTransition />` router allowed for smooth transitions between pages, but any stateful islands or elements you might have had would be rerendered.
```
Comment (sarah11918) [inline src/content/blog/astro-2100.mdx:30]:
```suggestion
Our first release of the `<ViewTransition />` router brought smooth transitions between pages, but still caused any stateful islands or elements to be rerendered.
```

File: src/content/blog/astro-2100.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+In 2.10 we have fixed this with a new `transition:persist` directive.
```
Comment (sarah11918) [inline src/content/blog/astro-2100.mdx:32]:
```suggestion
In 2.10, we've added a new `transition:persist` directive to the API to address this.
```
"fix" is a super strong claim at this stage of the game, maybe?  :sweat_smile:

File: src/content/blog/astro-2100.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+With this directive, you can keep the state of DOM elements and islands on the old page when transitioning to the new page. [Here's a demo](https://stackblitz.com/edit/github-stfdbe-ikkp2e?file=src%2Fpages%2Findex.astro) of it in action.
```
Comment (sarah11918) [inline src/content/blog/astro-2100.mdx:34]:
```suggestion
With this directive, you can keep the state of DOM elements and islands from the old page when transitioning to the new page. [Here's a demo](https://stackblitz.com/edit/github-stfdbe-ikkp2e?file=src%2Fpages%2Findex.astro) of it in action.
```
This *could* be read as "keeping stuff ON the last page"... to rephrase like retaining the stuff FROM the last page when you go TO the new page is just less ambiguous!

File: src/content/blog/astro-2100.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+```astro
+<Counter count={5} client:load transition:persist />
+```
+
+See our [View Transitions Guide](https://docs.astro.build/en/guides/view-transitions/#maintaining-state) to learn more on usage.
+
+## First Look at Astro 3.0
+
+We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. For those adventurous you can try out the release and see if you have compatibility issues with your app. So far this release contains mostly deprecations and removals of very old patterns. A few changes you might be interested in include:
```
Comment (sarah11918) [inline src/content/blog/astro-2100.mdx:59]:
```suggestion
We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. If you're feeling adventurous, try it today! compatibility issues with your app. So far this release contains mostly deprecations and removals of very old patterns. However, you may experience some compatibility issues or breaking changes.

Some noteworthy changes include:
```

File: src/content/blog/astro-2100.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+- HTML is now compressed by default.
+- API routes now used capitalized method names, for example: `export function GET() ...`
+- Adapters can inform Astro of which features they support, such as splitting entrypoints per page and Sharp / Squoosh compatibility.
+
+[See the changelog](https://github.com/withastro/astro/blob/refs/heads/next/packages/astro/CHANGELOG.md#300-beta0) for a full list of the changes. We currently do not have documentation for the breaking changes but are working on a guide. As we approach 3.0 there will be more information about the changes.
+
+Once all of the breaking changes are complete there will be a release candidate and we hope to release the stable 3.0 in late August / early September.
```
Comment (sarah11918) [inline src/content/blog/astro-2100.mdx:67]:
```suggestion
You can expect a release candidate once all of the breaking changes are complete, and we hope to release the stable 3.0 in late August / early September.
```

**Review by sarah11918 — COMMENTED** [review]
Looks great Matthew! Just a few suggestions from me!

File: src/content/blog/astro-2100.mdx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+coverImage: "/src/content/blog/_images/astro-2100/blog-hero-2_10-release.png"
+socialImage: "/src/content/blog/_images/astro-2100/og-image-2_10-release.png"
+lang: "en"
+---
+
+Astro 2.10 is out with improvements to our experimental View transitions support. Additionally we've released our first early beta of Astro 3.0. Read on to learn more!
+
+
+- **[Persistent UI in View transitions](#view-transitions-experimental)**
+- **[First Look at Astro 3.0](#astro-3)**
```
[the remark this answers is not in the document]
Comment (matthewp) [reply src/content/blog/astro-2100.mdx:16]: Oops, I need to check both of these.

File: src/content/blog/astro-2100.mdx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+While you're at it, upgrade any `@astrojs/*` integrations and adapters you have installed, too!
+
+## Persistent UI in View Transitions
+
+Our first release of the `<ViewTransition />` router brought smooth transitions between pages, but still caused any stateful islands or elements to be rerendered.
```
Comment (Yan-Thomas) [inline src/content/blog/astro-2100.mdx:30]: Comma in compound predicate isn't necessary, also missing hyphen in `rerendered`
```suggestion
Our first release of the `<ViewTransition />` router brought smooth transitions between pages but still caused any stateful islands or elements to be re-rendered.
```

File: src/content/blog/astro-2100.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (matthewp) [reply src/content/blog/astro-2100.mdx]: Should be fixed.

File: src/content/blog/astro-2100.mdx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+```astro
+<Counter count={5} client:load transition:persist />
+```
+
+See our [View Transitions Guide](https://docs.astro.build/en/guides/view-transitions/#maintaining-state) to learn more on usage.
+
+## First Look at Astro 3.0
+
+We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. If you're feeling adventurous, try it today! compatibility issues with your app. So far this release contains mostly deprecations and removals of very old patterns. However, you may experience some compatibility issues or breaking changes.
```
Comment (sarah11918) [inline src/content/blog/astro-2100.mdx:59]:
```suggestion
We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. If you're feeling adventurous, try it today! So far this release contains mostly deprecations and removals of very old patterns. However, you may experience some compatibility issues or breaking changes. Be warned, this early preview is changing rapidly!
```

**Review by sarah11918 — APPROVED** [review]
Good to go!

**Pull request merged**